lib/spack/spack/reporters/cdash.py

```diff
@@ -142,27 +169,36 @@ def clean_log_event(event):
                     report_data[phase]['warnings'].append(
                         clean_log_event(warning))
 
+            if phase == 'update':
+                report_data[phase]['revision'] = self.revision
+
             # Write the report.
             report_name = phase.capitalize() + ".xml"
             phase_report = os.path.join(filename, report_name)
 
             with codecs.open(phase_report, 'w', 'utf-8') as f:
                 env = spack.tengine.make_environment()
-                site_template = os.path.join(self.template_dir, 'Site.xml')
-                t = env.get_template(site_template)
-                f.write(t.render(report_data))
+                if phase != 'update':
+                    # Update.xml stores site information differently
+                    # than the rest of the CTest XML files.
+                    site_template = os.path.join(self.template_dir, 'Site.xml')
+                    t = env.get_template(site_template)
+                    f.write(t.render(report_data))
 
                 phase_template = os.path.join(self.template_dir, report_name)
                 t = env.get_template(phase_template)
                 f.write(t.render(report_data))
             self.upload(phase_report)
+        self.print_cdash_link()
 
     def concretization_report(self, filename, msg):
         report_data = {}
         self.initialize_report(filename, report_data)
-        report_data['starttime'] = self.starttime
-        report_data['endtime'] = self.starttime
-        report_data['msg'] = msg
+        report_data['update'] = {}
+        report_data['update']['starttime'] = self.endtime
```

Comment:
Why we don't have a start time for concretization?

Reply:
I could be mistaken, but I don't think the CDash reporter has access to how long it took Spack to concretization the spec(s). So instead we report a duration of zero seconds for the "Update" step.